ast/term.go

```diff
@@ -185,15 +185,20 @@ func ValueToInterface(v Value, resolver Resolver) (interface{}, error) {
 			if err != nil {
 				return err
 			}
-			asStr, stringKey := ki.(string)
-			if !stringKey {
-				return fmt.Errorf("object value has non-string key (%T)", ki)
+			var str string
+			var ok bool
+			if str, ok = ki.(string); !ok {
+				var buf bytes.Buffer
+				if err := json.NewEncoder(&buf).Encode(ki); err != nil {
+					return err
+				}
+				str = strings.TrimSpace(buf.String())
 			}
 			vi, err := ValueToInterface(v.Value, resolver)
 			if err != nil {
 				return err
 			}
-			buf[asStr] = vi
+			buf[str] = vi
```

Comment:
In case of a clash (`{"1":0, 1:0}`), the key processed second will overwrite the one processed first, right?

I see no solid solution here, only ways to make it less likely, like marshaling the example thing to `{"[1]":0,"1":0}`.

Reply:
Tried it briefly to verify:

```console
> { 1: true, "1": false, "{}": "foo", {}: "bar" }
{
  "1": false,
  "{}": "bar"
}
>
```

Reply:
Nice find. Agreed, there's no good solution here. I think this is still better than having policy evaluation return an error. I'm going to go ahead and merge.